docs/setup-cluster/deploy-cluster/firewall-rules.rst

```diff
@@ -0,0 +1,55 @@
+:orphan:
+
+.. _firewall-rules:
+
+################
+ Firewall Rules
+################
+
+The firewall rules must satisfy the following network access requirements for the master and agents.
+
+********
+ Master
+********
+
+-  Inbound TCP to the master's network port from the Determined agent instances, as well as all
+   machines where developers want to use the Determined CLI or WebUI. The default port is ``8443``
+   if TLS is enabled and ``8080`` if not.
+
+-  Outbound TCP to all ports on the Determined agents.
+
+********
+ Agents
+********
+
+-  Inbound TCP from all ports on the master to all ports on the agent.
+
+-  Outbound TCP from all ports on the agent to the master's network port.
+
+-  Outbound TCP to the services that host the Docker images, packages, data, and other resources
+   that need to be accessed by user workloads. For example, if your data is stored on Amazon S3,
+   ensure the firewall rules allow access to this data.
+
+-  Inbound and outbound TCP on all ports to and from each Determined agent. The details are as
+   follows:
+
+   -  Inbound and outbound TCP ports 1734 and 1750 are used for synchronization between trial
+      containers.
+   -  Inbound and outbound TCP port 12350 is used for internal SSH-based communication between trial
+      containers.
+   -  When using ``DeepSpeedTrial``, port 29500 is used by for rendezvous between trial containers.
+   -  When using ``PyTorchTrial`` with the "torch" distributed training backend, port 29400 is used
+      for rendezvous between trial containers
+   -  For all other distributed training modes, inbound and outbound TCP port 12355 is used for GLOO
+      rendezvous between trial containers.
+   -  Inbound and outbound ephemeral TCP ports in the range 1024-65536 are used for communication
+      between trials via GLOO.
+   -  For every GPU on each agent machine, an inbound and outbound ephemeral TCP port in the range
+      1024-65536 is used for communication between trials via NCCL.
+   -  Two additional ephemeral TCP ports in the range 1024-65536 are used for additional intra-trial
+      communication between trial containers.
+   -  Each TensorBoard uses a port in the range 2600–2899
+   -  Each notebook uses a port in the range 2900–3199
+   -  Each shell uses a port in the range 3200–3599
```

Comment:
This feels like pointless documentation to me (yeah, I know you didn't write it).

The reasons why I think it's pointless are:
- We start by saying we require all ports to be open between agents, so the details are pointless
- As a user, I would assume the details mean "if you really want you can open only the following ports" but in reality all these ports are required since we specify that we allow any ephimeral port for each gpu.
- Why on earth do we specify that we use any ephimeral port.  That's the default behavior of all software ever.  (I feel the same about "outbound traffic from ports x", by the way).
- These ports are a bit out-of-date anyway.
- The ranges specified for tensorboards, networks, and shells don't take into account effects cause by bridge networking.

I'd like a second opinion from perhaps @dannysauer or @loksonarius but this sort of docs seems nonstandard and I'd rather we just delete it.

Reply:
These docs seem a bit over detailed for install instructions. That's said, specifics on what ports are used for what and how is a pretty standard if not nice to have as a consumer.

Reply:
I like the idea of documenting ports we actually expect to use and what they're used for.  Even if users aren't setting up a firewall between hosts, it's helpful to me as a sysadmin to have some idea what to expect. Specifically: sometimes there are things like locally-installed management tools using a well-known port from the user port range (1024-49151), and it's helpful to know that there may be a conflict.  I don't know if that's the target audience for this doc, but if it is, I'd like to retain documentation of an up-to-date list of ports we actually do expect.  Ideally this would be free of conflicts with [reserved ports](http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml).  If we can't provide accurate documentation of actual ports in use, then this document is probably more harmful than beneficial.  Like Ryan says in his last couple of bullets, if this list is obsolete and going to be useless

That said, I very much do not like specifying the full range of ephemeral ports either way.

> 1024-65536 is used for communication between trials via NCCL. 

That (and the other instance) should be removed or at least edited, as 1) that is *not* the range of ephemeral ports on Linux (it's 32768-61000 on Linux, and [other ranges](https://en.wikipedia.org/wiki/Ephemeral_port) on some other platforms) and 2) the range could be changed on a per-site basis.  If someone doesn't know what the ephemeral ports are on their system, giving them a range helps nothing.

Reply:
resolved via slack

replaced bullet points that referenced `1024-65536` with 

> Trials additionally use OS-selected ephemeral ports for communication via GLOO and NCCL.